File: ios/Classes/KsFlutterEvent.m
//
//  KsFlutterEvent.m
//  ks_ads_flutter
//
//  Created by 爱互动 on 2021/9/13.
//

#import <Foundation/Foundation.h>
#import <Flutter/Flutter.h>
#import "KsFlutterEvent.h"

@implementation KsFlutterEvent {
    FlutterEventSink eventSink;
}

- (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
    FlutterEventChannel *eventChannel = [FlutterEventChannel eventChannelWithName:@"com.ahd.ks_ads/ad_event" binaryMessenger:registrar.messenger];
    [eventChannel setStreamHandler:self];
    return self;
}

- (FlutterError * _Nullable)onCancelWithArguments:(id _Nullable)arguments {
    eventSink = nil;
    return nil;
}

- (FlutterError * _Nullable)onListenWithArguments:(id _Nullable)arguments eventSink:(nonnull FlutterEventSink)events {
    eventSink = events;
    return nil;
}

- (void)sendEvent:(NSDictionary *)event {
    eventSink(event);
}

@end

File: android/src/main/java/com/ahd/ks_ads_flutter/KsFlutterEvent.java
package com.ahd.ks_ads_flutter;

import androidx.annotation.NonNull;

import java.util.Map;

import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.plugin.common.EventChannel;

public class KsFlutterEvent implements FlutterPlugin, EventChannel.StreamHandler {

    private static final KsFlutterEvent instance = new KsFlutterEvent();

    public static KsFlutterEvent getInstance() {
        return instance;
    }

    private EventChannel eventChannel;
    private EventChannel.EventSink eventSink;

    @Override
    public void onAttachedToEngine(@NonNull FlutterPluginBinding binding) {
        eventChannel = new EventChannel(binding.getBinaryMessenger(), "com.ahd.ks_ads/ad_event");
        eventChannel.setStreamHandler(this);
    }

    @Override
    public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
        eventChannel.setStreamHandler(null);
        eventChannel = null;
    }

    @Override
    public void onListen(Object arguments, EventChannel.EventSink events) {
        eventSink = events;
    }

    @Override
    public void onCancel(Object arguments) {
        eventSink = null;
    }

    public void sendEvent(Map<String, Object> content) {
        if (eventSink != null) {
            eventSink.success(content);
        }
    }
}
